parsl/dataflow/dflow.py

```diff
@@ -729,6 +729,8 @@ def submit(self, func, app_args, executors='all', fn_hash=None, cache=False, ign
                                     kw)
                     )
 
+        resource_specification = app_kwargs.pop('parsl_resource_specification', {})
```

Comment:
Other references to parsl-specific keyword args use `get` or `[]`, rather than `pop` which removes `parsl_resource_specification` from the dictionary.

The behaviour should be consistent unless there is a clear reason to make the behaviour different.

Reply:
It is `pop` here because `parsl_resource_specification` cant be a kwarg passed to the app---it will raise an exception of unknown kwarg

Reply:
> parsl_resource_specification cant be a kwarg passed to the app---otherwise it will raise an exception of unknown kwarg

That is the same behaviour as `stdout`, `stderr` and `label`. What is the story to tell a user that `label` or `stdout` needs to be declared, but `parsl_resource_specification` doesn't?

Reply:
I see your point now. I changed to `get`